### PR TITLE
[IMP] mail: add test tour for guest pages

### PR DESCRIPTION
**Pull request by sdegueldre**

### CODE REVIEW

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

File: addons/mail/tests/__init__.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import test_mail_channel
+from . import test_mail_channel_as_guest
```
Comment (tde-banana-odoo) [inline addons/mail/tests/__init__.py:5]: Generally, we use a specific file naming for UI tests, like ``test_mail_channel_ui`` . Could be ``test_discuss_ui`` probably, if you want to include some discuss tours.

File: addons/mail/tests/test_mail_channel_as_guest.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import odoo
+from odoo.tests import HttpCase
+
+@odoo.tests.tagged('-at_install', 'post_install')
+class TestMailGuestPages(HttpCase):
+    def test_mail_channel_as_guest(self):
+        """Checks that the invite page redirects to the channel and that all
+        modules load correctly on the welcome and channel page"""
+        channel = self.env['mail.channel'].create({
+            'name': 'Test channel',
```
Comment (tde-banana-odoo) [inline addons/mail/tests/test_mail_channel_as_guest.py:13]: Probably a bit short. Shouldn't we specifically target public channels, ... ?

**Review by tde-banana-odoo — COMMENTED** [review]
SM review: linting / guidelines

File: addons/mail/tests/test_mail_channel_as_guest.py (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/tests/test_mail_channel_as_guest.py:13]: The main use case is actually private channels (group created from "start a meeting"). But all channels are available through invitation link (except DM), so the type or visibility is not really a factor for the current test. We just want to have at least one test to ensure the page even loads at all, as it's doing a lot of custom booting code that will definitely be broken by changes in assets bundles if people don't know it exists.

**Pull request closed without merging**